src/Driver/PDO/SQLite/Connection.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Driver\PDO\SQLite;
+
+use Doctrine\DBAL\Driver\Exception\SequencesNotSupported;
+use Doctrine\DBAL\Driver\PDO\Connection as PDOConnection;
+use Doctrine\DBAL\Driver\Result;
+use Doctrine\DBAL\Driver\ServerInfoAwareConnection;
+use Doctrine\DBAL\Driver\Statement as StatementInterface;
+
+final class Connection implements ServerInfoAwareConnection
+{
+    /** @var PDOConnection */
+    private $connection;
+
+    public function __construct(PDOConnection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function prepare(string $sql): StatementInterface
+    {
+        return $this->connection->prepare($sql);
+    }
+
+    public function query(string $sql): Result
+    {
+        return $this->connection->query($sql);
+    }
+
+    public function quote(string $value): string
+    {
+        return $this->connection->quote($value);
+    }
+
+    public function exec(string $sql): int
+    {
+        return $this->connection->exec($sql);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function lastInsertId()
+    {
+        return $this->connection->lastInsertId();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getSequenceNumber(string $name)
+    {
+        // SQLite does not support sequences. However, PDO::lastInsertId() ignores the name parameter, and returns
+        // the last insert ID even if a sequence name is given. We expect an exception in that case.
+        throw SequencesNotSupported::new();
+    }
```

Comment:
Do we really want to make `Driver\PDO\Connection` final? Because of this, we have to create decorator classes just to make `lastInsertId()` or `getSequenceNumber()` throw, with all other methods just proxying to the decorated driver connection.

Sure, we could create an `AbstractConnectionDecorator`, but then, why override the decorator's method rather than the Connection's method directly?

Another option is to not create driver-specific connection classes for this purpose, but then we end up with [this](https://github.com/doctrine/dbal/pull/4324/files#diff-d4cd4042dbac00d3f1359475b4168d72R99-R103), which I doubt is good design.

Reply:
I don't see much problem in creating a decorator.

Reply:
> I don't see much problem in creating a decorator.

Do you mean an abstract decorator, or a decorator for each Connection class?

Creating a decorator for each Connection class looks cumbersome to me, and creates too much distraction when reading the Connection classes' code: you can't clearly separate the custom logic from the boilerplate logic. 

On the other hand, if your suggestion is to create an abstract decorator: what's the point of making a class `final` to then create an abstract decorator to extend from? That's an extra class for... nothing if you ask me.

Reply:
My suggestion is to create a class for each connector. It may look cumbersome because PHP as a language doesn't make it straightforward, not because of the composition approach itself.

Reply:
I'm not convinced, but I'll leave the decorator there, then. If you happen to change your mind, we can remove `final` from the  `Driver\PDO\Connection`.